### PR TITLE
【Hackathon 5th No.19】Add ContinuousBernoulli and MultivariateNormal API

### PR DESCRIPTION
**Pull request by NKNaN**
### PR types
New features

### PR changes
APIs

### Description
Add ContinuousBernoulli and MultivariateNormal API
- rfc of ContinuousBernoulli: https://github.com/PaddlePaddle/community/blob/master/rfcs/APIs/20230927_api_design_for_ContinuousBernoulli.md
- rfc of MultivariateNormal: https://github.com/PaddlePaddle/community/blob/master/rfcs/APIs/20230927_api_design_for_MultivariateNormal.md

### CODE REVIEW

**Comment by NKNaN** [comment]
### Math Derivation for entropy of the Continuous Bernoulli distribution and kl_divergence of 2 Continuous Bernoulli distributions:

- entropy:
```math
\begin{aligned}
H &= -\int_x C(\lambda) \lambda^x (1-\lambda)^{1-x} \log\{C(\lambda) \lambda^x (1-\lambda)^{1-x}\} dx \\
& = -\int_0^1 C \lambda^x (1-\lambda)^{1-x} \left[ \log C + x \log \lambda + (1 -x) \log (1 - \lambda)\right] dx \\
& = -\left[ C \log C \int_0^1 \lambda^x (1-\lambda)^{1-x} dx + C \log \lambda \int_0^1 x \lambda^x (1-\lambda)^{1-x} + C \log(1 - \lambda) \int_0^1 (1-x) \lambda^x (1-\lambda)^{1-x} \right] \\
& = - \left[ \log C + \mathbb{E}(X) \log \lambda + \mathbb{E}(1 - X) \log(1 - \lambda) \right] \\
& = -\log C + \left[ \log (1 - \lambda) -\log \lambda \right] \mathbb{E}(X)  - \log(1 - \lambda) 
\end{aligned}
```
- kl_divergence:
```math
\begin{aligned}
\mathcal{D}_{KL}(p_1|| p_2) &= \int_x p_1(x)\log\frac{p_1(x)}{p_2(x)} dx \\
& = \int_0^1 C_1 \lambda_1^x (1-\lambda_1)^{1-x} \{\log[C_1 \lambda_1^x (1-\lambda_1)^{1-x}] - \log[C_2 \lambda_2^x (1-\lambda_2)^{1-x}]\} dx \\
& = -H - [C_1 \log C_2 \int_0^1 \lambda_1^x (1-\lambda_1)^{1-x} dx + C_1 \log \lambda_2 \int_0^1 x \lambda_1^x (1-\lambda_1)^{1-x} dx + C_1 \log (1-\lambda_2) \int_0^1 (1-x) \lambda_1^x (1-\lambda_1)^{1-x} dx] \\
& = -H - [\log C_2 + \log \lambda_2 \mathbb{E}_1(X) + \log (1-\lambda_2)  \mathbb{E}_1(1-X)  ] \\
& = - H - \{\log C_2 + [\log \lambda_2 -  \log (1-\lambda_2)]  \mathbb{E}_1(X) +  \log (1-\lambda_2)  \}
\end{aligned}
```

**Comment by NKNaN** [comment]
### Math Derivation for entropy of the Multivariate Normal distribution and kl_divergence of 2 Multivariate Normal distributions:

- entropy:
```math
\begin{aligned}
H &= -\int_x f(x) \log f(x) dx \\
& = -\int_{x \in \mathbb{R}^n} f(x) \{ -\frac{n}{2}\log(2\pi) -\frac{1}{2} (x-\mu)^{\intercal} \Sigma^{-1} (x-\mu) - \frac{1}{2}\log (\det\Sigma) \} dx \\
& = -\int_{x \in \mathbb{R}^n} f(x) \{ -\frac{n}{2}\log(2\pi) -\frac{1}{2} [A^{-1}(x-\mu)]^{\intercal}[A^{-1}(x-\mu)] - \log (\det A) \} dx \\
& = \frac{n}{2} \log(2\pi) + \log {\det A} + \frac{1}{2}\int_{x \in \mathbb{R}^n}   [A^{-1}(x-\mu)]^{\intercal}[A^{-1}(x-\mu)] f(x) dx\\
& = \frac{n}{2} \log(2\pi) + \log {\det A} +  \frac{1}{2} \mathbb{E}[(X-\mu)^{\intercal} \Sigma^{-1} (X - \mu)] \\
& =  \frac{n}{2} \log(2\pi) + \log {\det A} +  \frac{1}{2} \mathbb{E}[tr[(X-\mu)^{\intercal} \Sigma^{-1} (X - \mu)] ] \\
& =  \frac{n}{2} \log(2\pi) + \log {\det A} +  \frac{1}{2}  \mathbb{E}[tr[\Sigma^{-1} (X - \mu) (X-\mu)^{\intercal}]] \\
& = \frac{n}{2} \log(2\pi) + \log {\det A} +  \frac{1}{2} tr[\mathbb{E}[\Sigma^{-1} (X - \mu) (X-\mu)^{\intercal}]] \\
& = \frac{n}{2} \log(2\pi) + \log {\det A} +  \frac{1}{2} tr[\Sigma^{-1} \mathbb{E}[ (X - \mu) (X-\mu)^{\intercal}]] \\
& =  \frac{n}{2} \log(2\pi) + \log {\det A} + \frac{1}{2} tr[\Sigma^{-1} \Sigma] \\
& = \frac{n}{2} \log(2\pi) + \log {\det A} + \frac{n}{2}
\end{aligned}
```
- kl_divergence:
```math
\begin{aligned}
\mathcal{D}_{KL}(f_1|| f_2) &= \int_x f_1(x)\log\frac{f_1(x)}{f_2(x)} dx \\
& = \int_{x \in \mathbb{R}^n} f_1(x)\left\{\left[ -\frac{n}{2} \log(2\pi) - \log(\det A_1) - \frac{1}{2}(x-\mu_1)^{\intercal} \Sigma_1^{-1} (x - \mu_1) \right] + \left[ \frac{n}{2} \log(2\pi) + \log(\det A_2) + \frac{1}{2}(x-\mu_2)^{\intercal} \Sigma_21^{-1} (x - \mu_2)\right]\right\} dx \\
& = \log(\det A_2) - \log(\det A_1) +\frac{1}{2}\mathbb{E}_1[(X-\mu_2)^{\intercal} \Sigma_2^{-1} (X - \mu_2)] -\frac{n}{2} \\
& = \log(\det A_2) - \log(\det A_1) +\frac{1}{2}tr [\Sigma_2^{-1}\mathbb{E}_1[ (X - \mu_2) (X-\mu_2)^{\intercal} ]] -\frac{n}{2} \\
& = \log(\det A_2) - \log(\det A_1) -\frac{n}{2} +\frac{1}{2}tr [\Sigma_2^{-1}\mathbb{E}_1[ XX^{\intercal} -X \mu_2^{\intercal} - \mu_2 X^{\intercal} + \mu_2\mu_2^{\intercal}]] \\
& = \log(\det A_2) - \log(\det A_1) -\frac{n}{2} +\frac{1}{2}tr [\Sigma_2^{-1} [ Var_1(X) + \mathbb{E}_1(X)\mathbb{E}_1(X)^{\intercal} -\mu_1\mu_2^{\intercal} - \mu_2 \mu_1^{\intercal} + \mu_2\mu_2^{\intercal}]] \\
& = \log(\det A_2) - \log(\det A_1) -\frac{n}{2} +\frac{1}{2}tr [\Sigma_2^{-1} [ \Sigma_1 + \mu_1\mu_1^{\intercal} -\mu_1\mu_2^{\intercal} - \mu_2 \mu_1^{\intercal} + \mu_2\mu_2^{\intercal}]] \\
& =  \log(\det A_2) - \log(\det A_1) -\frac{n}{2} +\frac{1}{2}tr [\Sigma_2^{-1} \Sigma_1 + (\mu_1 - \mu_2)^{\intercal} \Sigma_2^{-1}  (\mu_1 - \mu_2)] \\
& =  \log(\det A_2) - \log(\det A_1) -\frac{n}{2} +\frac{1}{2}[tr [\Sigma_2^{-1} \Sigma_1] + (\mu_1 - \mu_2)^{\intercal} \Sigma_2^{-1}  (\mu_1 - \mu_2)] \\
\end{aligned}
```

**Comment by luotao1** [comment]
`test_distribution_continuous_bernoulli_static` 单测没过

File: python/paddle/distribution/continuous_bernoulli.py [hunk 1]
```diff
@@ -0,0 +1,435 @@
...
+    def _to_tensor(self, probability):
+        """Convert the input parameters into tensors with the global default dtype
+
+        Returns:
+            Tensor: converted probability.
+        """
+        # convert type
+        if isinstance(probability, (float, int)):
+            probability = [probability]
+        probability = paddle.to_tensor(probability, dtype=self.dtype)
```
Comment (cxxly) [inline python/paddle/distribution/continuous_bernoulli.py]: 如果probability本身是Tensor，这里会改变probability的数据类型。别入用户传入p是fp32, 默认数据类型是fp64

File: python/paddle/distribution/multivariate_normal.py [hunk 2]
```diff
@@ -0,0 +1,499 @@
...
+        self,
+        loc,
+        covariance_matrix=None,
+        precision_matrix=None,
+        scale_tril=None,
+    ):
+        self.dtype = paddle.get_default_dtype()
+        if isinstance(loc, (float, int)):
+            loc = [loc]
+        loc = paddle.to_tensor(loc, dtype=self.dtype)
```
Comment (cxxly) [inline python/paddle/distribution/multivariate_normal.py]: 同上，可能会有数据类型转换问题

File: test/distribution/test_distribution_continuous_bernoulli.py [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+        (
+            'one-dim',
+            parameterize.xrand((1,), min=0.0, max=1.0).astype("float32"),
+        ),
+        (
+            'multi-dim',
+            parameterize.xrand((2, 3), min=0.0, max=1.0).astype("float32"),
+        ),
+    ],
+)
```
Comment (cxxly) [inline test/distribution/test_distribution_continuous_bernoulli.py:174]: 测下不同数据类型

File: test/distribution/test_distribution_continuous_bernoulli.py [hunk 4]
```diff
@@ -0,0 +1,320 @@
...
+
+    def test_mean(self):
+        mean = self._dist.mean
+        self.assertEqual(mean.numpy().dtype, self.probability.dtype)
+        np.testing.assert_allclose(
+            mean,
+            self._np_dist.np_mean(),
+            rtol=0.00,
+            atol=0.20,
+        )
```
Comment (cxxly) [inline test/distribution/test_distribution_continuous_bernoulli.py:190]: 这个阈值设置标准是否有参考？感觉太大了

File: test/distribution/test_distribution_continuous_bernoulli.py [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+
+    def test_variance(self):
+        var = self._dist.variance
+        self.assertEqual(var.numpy().dtype, self.probability.dtype)
+        np.testing.assert_allclose(
+            var,
+            self._np_dist.np_variance(),
+            rtol=0.00,
+            atol=0.20,
+        )
```
Comment (cxxly) [inline test/distribution/test_distribution_continuous_bernoulli.py:200]: 同上

File: test/distribution/test_distribution_continuous_bernoulli.py [hunk 6]
```diff
@@ -0,0 +1,320 @@
...
+
+    def test_entropy(self):
+        entropy = self._dist.entropy()
+        self.assertEqual(entropy.numpy().dtype, self.probability.dtype)
+        np.testing.assert_allclose(
+            entropy,
+            self._np_dist.np_entropy(),
+            rtol=0.00,
+            atol=0.20,
+        )
```
Comment (cxxly) [inline test/distribution/test_distribution_continuous_bernoulli.py:210]: 同上

File: test/distribution/test_distribution_continuous_bernoulli.py [hunk 7]
```diff
@@ -0,0 +1,320 @@
...
+        samples = self._dist.sample(sample_shape)
+        sample_mean = samples.mean(axis=0)
+        sample_variance = samples.var(axis=0)
+
+        np.testing.assert_allclose(
+            sample_mean, self._dist.mean, atol=0, rtol=0.20
+        )
+        np.testing.assert_allclose(
+            sample_variance, self._dist.variance, atol=0, rtol=0.20
+        )
```
Comment (cxxly) [inline test/distribution/test_distribution_continuous_bernoulli.py:237]: 同上

File: test/distribution/test_distribution_continuous_bernoulli.py [hunk 8]
```diff
@@ -0,0 +1,320 @@
...
+        kl0 = self._dist1.kl_divergence(self._dist2)
+        kl1 = self._np_dist1.np_kl_divergence(self._np_dist2)
+
+        self.assertEqual(tuple(kl0.shape), self._dist1.batch_shape)
+        self.assertEqual(tuple(kl1.shape), self._dist1.batch_shape)
+        np.testing.assert_allclose(kl0, kl1, rtol=0.1, atol=0.1)
+
+
+if __name__ == '__main__':
+    unittest.main(argv=[''], verbosity=3, exit=False)
```
Comment (cxxly) [inline test/distribution/test_distribution_continuous_bernoulli.py:385]: 代码中有多处异常判断，增加一些异常测试

File: test/distribution/test_distribution_multivariate_normal.py [hunk 9]
```diff
@@ -0,0 +1,252 @@
...
+            parameterize.xrand((2,), dtype='float32', min=1, max=2),
+            np.array([[2.0, 1.0], [1.0, 2.0]]),
+        ),
+        (
+            'multi-batch',
+            parameterize.xrand((2, 3), dtype='float32', min=-2, max=-1),
+            np.array([[4.0, 2.5, 2.0], [2.5, 3.0, 1.2], [2.0, 1.2, 4.0]]),
+        ),
+    ],
+)
```
Comment (cxxly) [inline test/distribution/test_distribution_multivariate_normal.py:46]: 同上，增加下不同数据类型测试，测试用例能覆盖到不同case

File: test/distribution/test_distribution_multivariate_normal.py [hunk 10]
```diff
@@ -0,0 +1,252 @@
...
+        samples = self._dist.sample(sample_shape)
+        sample_mean = samples.mean(axis=0)
+        sample_variance = samples.var(axis=0)
+
+        np.testing.assert_allclose(
+            sample_mean, self._dist.mean, atol=0.00, rtol=0.20
+        )
+        np.testing.assert_allclose(
+            sample_variance, self._dist.variance, atol=0.00, rtol=0.20
+        )
```
Comment (cxxly) [inline test/distribution/test_distribution_multivariate_normal.py:104]: 同上，注释阈值参考标准

**Comment by cxxly** [comment]
如果有该PR导致的随机挂问题，大概率是采样测试部分，采样数据量设置过小，可适当增加采样数据

File: python/paddle/distribution/continuous_bernoulli.py (hunk 1 above)
Comment (NKNaN) [reply python/paddle/distribution/continuous_bernoulli.py]: 已修改

File: python/paddle/distribution/multivariate_normal.py (hunk 2 above)
Comment (NKNaN) [reply python/paddle/distribution/multivariate_normal.py]: 已修改

File: test/distribution/test_distribution_continuous_bernoulli.py (hunk 3 above)
Comment (NKNaN) [reply test/distribution/test_distribution_continuous_bernoulli.py:174]: 已添加对float64的测试

File: test/distribution/test_distribution_continuous_bernoulli.py (hunk 4 above)
Comment (NKNaN) [reply test/distribution/test_distribution_continuous_bernoulli.py:190]: ContinuousBernoulli 在 p=0.5 附近的计算因为浮点精度问题会变得不稳定，所以要对不稳定的范围做一个限定，在不稳定范围内用泰勒展开近似计算就可以变得稳定。之前在这一块的范围默认设定太小了，导致测试的时候总会出现不稳定情况。现在默认设定调大了，mean, variance, prob, log_prob, cdf, icdf 的测试阈值设定为 rtol=0.001, atol=0; entropy, kl_divergence 的设定为 rtol=0.005, atol=0; sample 的设定为 rtol=0.02, atol=0.

File: test/distribution/test_distribution_continuous_bernoulli.py (hunk 8 above)
Comment (NKNaN) [reply test/distribution/test_distribution_continuous_bernoulli.py:385]: 已添加

File: test/distribution/test_distribution_multivariate_normal.py (hunk 9 above)
Comment (NKNaN) [reply test/distribution/test_distribution_multivariate_normal.py:46]: 已添加

File: test/distribution/test_distribution_multivariate_normal.py (hunk 10 above)
Comment (NKNaN) [reply test/distribution/test_distribution_multivariate_normal.py:104]: MVN的sample的阈值现在设定为 rtol = 0.10, atol = 0，是参考 test_distribution_normal 和 test_distribution_lognormal

**Review by cxxly — DISMISSED** [review]
LGTM

**Comment by jeff41404** [comment]
code is fine, please add link of rfc in description above

**Comment by jeff41404** [comment]
the design of `ContinuousBernoulli` in [rfc](https://github.com/PaddlePaddle/community/blob/master/rfcs/APIs/20230927_api_design_for_continuous_bernoulli.md?plain=1#L656) needs to be consistent with the code

**Comment by jeff41404** [comment]
the [rfc](https://github.com/PaddlePaddle/community/blob/master/rfcs/APIs/20221027_api_design_for_MultivariateNormal.md?plain=1#L634) of `MultivariateNormal` have same issue

**Comment by NKNaN** [comment]
> code is fine, please add link of rfc in description above

添加了rfc链接，rfc设计文档需要做一些修改，已提相应[pr](https://github.com/PaddlePaddle/community/pull/783)

**Comment by luotao1** [comment]
对应中文文档可以提上来

**Comment by NKNaN** [comment]
> 对应中文文档可以提上来

已提中文[pr](https://github.com/PaddlePaddle/docs/pull/6392)

又修改了一下对应的英文文档

File: python/paddle/distribution/continuous_bernoulli.py [hunk 11]
```diff
@@ -0,0 +1,446 @@
...
+        \end{aligned}
+        \right.
+
+    Args:
+        probability(int|float|Tensor): The probability of Continuous Bernoulli distribution between [0, 1],
+        which characterize the shape of the pdf. If the input data type is int or float, the data type of
+        `probability` will be convert to a 1-D Tensor the paddle global default dtype.
+        eps(float): Specify the bandwith of the unstable calculation region near 0.5. The unstable calculation region
+        would be [0.5 - eps, 0.5 + eps], where the calculation is approximated by talyor expansion. The
+        default value is 0.02.
```
Comment (sunzhongkai588) [inline python/paddle/distribution/continuous_bernoulli.py]:
```suggestion
    Args:
        probability(int|float|Tensor): The probability of Continuous Bernoulli distribution between [0, 1],
            which characterize the shape of the pdf. If the input data type is int or float, the data type of
            `probability` will be convert to a 1-D Tensor the paddle global default dtype.
        eps(float): Specify the bandwith of the unstable calculation region near 0.5. The unstable calculation region
            would be [0.5 - eps, 0.5 + eps], where the calculation is approximated by talyor expansion. The
            default value is 0.02.
```
对于 Args 下的每个参数，同一个参数的描述换行需要加下缩进

File: python/paddle/distribution/continuous_bernoulli.py [hunk 12]
```diff
@@ -0,0 +1,446 @@
...
+
+import paddle
+from paddle.distribution import distribution
+
+
+class ContinuousBernoulli(distribution.Distribution):
+    r"""The Continuous Bernoulli distribution with parameter: `probability` characterizing the shape of the density function.
+    The Continuous Bernoulli distribution is defined on [0, 1], and it can be viewed as a continuous version of the Bernoulli distribution.
+
+    [1] Loaiza-Ganem, G., & Cunningham, J. P. The continuous Bernoulli: fixing a pervasive error in variational autoencoders. 2019.
```
Comment (sunzhongkai588) [inline python/paddle/distribution/continuous_bernoulli.py]: 能不能直接贴上论文的连接？引用方式参考 [如何让文档相互引用](https://github.com/PaddlePaddle/docs/wiki/%E9%A3%9E%E6%A1%A8%E6%96%87%E6%A1%A3%E7%9B%B8%E4%BA%92%E5%BC%95%E7%94%A8)；

File: python/paddle/distribution/continuous_bernoulli.py [hunk 13]
```diff
@@ -0,0 +1,446 @@
...
+
+    Args:
+        probability(int|float|Tensor): The probability of Continuous Bernoulli distribution between [0, 1],
+        which characterize the shape of the pdf. If the input data type is int or float, the data type of
+        `probability` will be convert to a 1-D Tensor the paddle global default dtype.
+        eps(float): Specify the bandwith of the unstable calculation region near 0.5. The unstable calculation region
+        would be [0.5 - eps, 0.5 + eps], where the calculation is approximated by talyor expansion. The
+        default value is 0.02.
+
+    Examples:
```
Comment (sunzhongkai588) [inline python/paddle/distribution/continuous_bernoulli.py:58]: - 请参考 [代码示例写法](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/dev_guides/api_contributing_guides/api_docs_guidelines_cn.html#id9) ，需要在前面加上 `>>>` 或 `...`
- 另外，输出不要用注释的形式，请仔细参考 [代码示例书写规范](https://www.paddlepaddle.org.cn/documentation/docs/zh/develop/dev_guides/style_guide_and_references/code_example_writing_specification_cn.html)

File: python/paddle/distribution/continuous_bernoulli.py [hunk 14]
```diff
@@ -0,0 +1,446 @@
...
+
+        The entropy is
+
+        .. math::
+
+            \mathcal{H}(X) = -\log C + \left[ \log (1 - \lambda) -\log \lambda \right] \mathbb{E}(X)  - \log(1 - \lambda)
+
+        In the above equation:
+
+        * :math:\Omega: is the support of the distribution.
```
Comment (sunzhongkai588) [inline python/paddle/distribution/continuous_bernoulli.py]:
```suggestion
        * :math:`\Omega` is the support of the distribution.
```

File: python/paddle/distribution/kl.py
Comment (sunzhongkai588) [inline python/paddle/distribution/kl.py:1]: 这篇文档和 `continuous_bernoulli.py` 有同样的问题，不赘述了

File: python/paddle/distribution/continuous_bernoulli.py (hunk 12 above)
Comment (NKNaN) [reply python/paddle/distribution/continuous_bernoulli.py]: 已修改

File: python/paddle/distribution/continuous_bernoulli.py (hunk 11 above)
Comment (NKNaN) [reply python/paddle/distribution/continuous_bernoulli.py]: 已修改

File: python/paddle/distribution/continuous_bernoulli.py (hunk 13 above)
Comment (NKNaN) [reply python/paddle/distribution/continuous_bernoulli.py:58]: 已修改

File: python/paddle/distribution/continuous_bernoulli.py (hunk 14 above)
Comment (NKNaN) [reply python/paddle/distribution/continuous_bernoulli.py]: 已修改

File: python/paddle/distribution/kl.py
Comment (NKNaN) [reply python/paddle/distribution/kl.py:1]: 对应处已修改

**Comment by cxxly** [comment]
抱歉，再补充个Comment，ContinuousBernoulli 签名和PyTorch保持一致

**Comment by NKNaN** [comment]
> 抱歉，再补充个Comment，ContinuousBernoulli 签名和PyTorch保持一致

已修改

**Review by sunzhongkai588 — DISMISSED** [review]
LGTM～

doc-preview CI 中发现新的 system message 错误， @ooooo-create 之后全量的再检查一遍相关错误并修复叭

File: python/paddle/distribution/continuous_bernoulli.py [hunk 15]
```diff
@@ -0,0 +1,454 @@
...
+            [0.05623737, 0.10000000])
+
+            >>> rv1 = ContinuousBernoulli(paddle.to_tensor([0.2, 0.8]))
+            >>> rv2 = ContinuousBernoulli(paddle.to_tensor([0.7, 0.5]))
+            >>> print(rv1.kl_divergence(rv2))
+            Tensor(shape=[2], dtype=float32, place=Place(cpu), stop_gradient=True,
+            [0.20103608, 0.07641447])
+    """
+
+    def __init__(self, probs=None, lims=(0.499, 0.501)):
```
Comment (cxxly) [inline python/paddle/distribution/continuous_bernoulli.py]: 不用加None，probs是必选参数

Comment (NKNaN) [reply python/paddle/distribution/continuous_bernoulli.py]: 已修改

**Review by cxxly — APPROVED** [review]
LGTM

**Comment by jeff41404** [comment]
There are spelling and capitalization issues in the link of rfc, eg should be `20230927_api_design_for_ContinuousBernoulli.md` instead of `20230927_api_design_for_continuous_bernoulli.md`, cause error of 404

**Comment by jeff41404** [comment]
The [rfc of MultivariateNormal](https://github.com/PaddlePaddle/community/blob/master/rfcs/APIs/20221027_api_design_for_MultivariateNormal.md?plain=1#L626-L627) lacks an introduction to parameter `precision_matrix` and `scale_tril`

**Comment by NKNaN** [comment]
> There are spelling and capitalization issues in the link of rfc, eg should be `20230927_api_design_for_ContinuousBernoulli.md` instead of `20230927_api_design_for_continuous_bernoulli.md`, cause error of 404

Links have been updated

**Review by jeff41404 — APPROVED** [review]
LGTM

**Pull request merged**